Lib/unittest/test/testmock/testmock.py

```diff
@@ -1891,6 +1917,16 @@ def foo(a, b): pass
 
         self.assertRaises(TypeError, mock.child, 1)
         self.assertEqual(mock.mock_calls, [call.child(1, 2)])
+        self.assertIn('mock.child', repr(mock.child.mock))
+
+        parent = Mock()
+        parent.attach_mock(create_autospec(foo, name='bar'), 'child')
+        parent.child(1, 2)
```

Comment:
Why are we calling `child` direct here?

Reply:
Sorry, I don't get the comment here. The foo function is attached as a `child` attribute so I am calling `parent.child` to ensure parent records the call. I am not getting the direct call part here.

Reply:
I guess I don't really understand the use case for `attach_mock` so happy to go with what you say.